### PR TITLE
frontend/feat:#2559 added-wasOfferedAssistance-in-feedbackReq-in-purple-rides

### PR DESCRIPTION
**Pull request by Shikhar-24**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added wasOfferedAssistance in feedbackReq in purple rides
Backend changes : [link](https://github.com/nammayatri/nammayatri/pull/3213)
[video link](https://drive.google.com/file/d/1XHnY_J54LZjTR0RuH0U5f0mwHN11lNpm/view?usp=drive_link)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 1]
```diff
@@ -1124,12 +1124,13 @@ rideCompletedCardConfig state = let
           issueFaced = state.data.ratingViewState.issueFacedView,
           selectedYesNoButton = state.data.ratingViewState.selectedYesNoButton,
           reportIssuePopUpConfig = reportIssuePopUpConfig state,
-          title = (getString DID_YOU_FACE_ANY_ISSUE),
-          subTitle = (getString WE_NOTICED_YOUR_RIDE_ENDED_AWAY),
+          title = if state.props.showOfferedAssistancePopUp then (getString DID_THE_DRIVER_OFFER_ASSISTANCE) else (getString DID_YOU_FACE_ANY_ISSUE),
+          subTitle = if state.props.showOfferedAssistancePopUp then "Was your driver understanding of your needs and accomodate them on the journey?" else (getString WE_NOTICED_YOUR_RIDE_ENDED_AWAY),
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:1128]: @Shikhar-24 Add translations for this

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 2]
```diff
@@ -190,9 +190,9 @@ skipButtonConfig state =
         , margin = MarginTop 22
         , id = "SkipButton"
         , enableLoader = (JB.getBtnLoader "SkipButton")
-        , visibility = if not issueFaced || state.data.ratingViewState.doneButtonVisibility then VISIBLE else GONE
-        , isClickable = issueFaced || state.data.ratingViewState.selectedRating > 0
-        , alpha = if issueFaced || (state.data.ratingViewState.selectedRating >= 1) then 1.0 else 0.4
+        , visibility = if state.data.ratingViewState.doneButtonVisibility || not state.props.showOfferedAssistancePopUp then VISIBLE else GONE
+        , isClickable = issueFaced || state.data.ratingViewState.selectedRating > 0 || state.data.ratingViewState.selectedYesNoButton > -1
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:194]: Can you revisit these all conditions 
Also Don't you think  `>=0` will be more clear than ` > -1`

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs [hunk 3]
```diff
@@ -121,7 +121,9 @@ initData = {
         issueFacedView : false,
         issueReason : Nothing,
         issueDescription : "",
-        rideBookingRes : dummyRideBooking
+        rideBookingRes : dummyRideBooking,
+        wasOfferedAssistance : Nothing,
+        donButtonActivate : false
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs:126]: Can you recheck this variable name ?

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs (hunk 1 above)
Comment (Shikhar-24) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs]: done

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs (hunk 2 above)
Comment (Shikhar-24) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs]: done

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs (hunk 3 above)
Comment (Shikhar-24) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs]: Removed as they are not needed, added them for done button visibility but then handled it differently.

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 4]
```diff
@@ -190,9 +190,9 @@ skipButtonConfig state =
         , margin = MarginTop 22
         , id = "SkipButton"
         , enableLoader = (JB.getBtnLoader "SkipButton")
-        , visibility = if not issueFaced || state.data.ratingViewState.doneButtonVisibility then VISIBLE else GONE
-        , isClickable = issueFaced || state.data.ratingViewState.selectedRating > 0
-        , alpha = if issueFaced || (state.data.ratingViewState.selectedRating >= 1) then 1.0 else 0.4
+        , visibility = if state.data.ratingViewState.doneButtonVisibility || not state.props.showOfferedAssistancePopUp then VISIBLE else GONE
+        , isClickable = issueFaced || state.data.ratingViewState.selectedRating > 0 || state.data.ratingViewState.selectedYesNoButton >= 0
+        , alpha = if issueFaced || (state.data.ratingViewState.selectedRating >= 1) || state.data.ratingViewState.selectedYesNoButton >= 0 then 1.0 else 0.4
```
[the remark this answers is not in the document]
Comment (Shikhar-24) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs]: done

File: Frontend/ui-customer/src/Flow.purs [hunk 5]
```diff
@@ -1063,7 +1064,7 @@ homeScreenFlow = do
     SUBMIT_RATING state -> do
       liftFlowBT $ logEventWithParams logField_ "ny_user_ride_give_feedback" "Rating" (show $ state.data.rating)
       _ <- Remote.bookingFeedbackBT (Remote.makeRideFeedBackReq (state.data.rideRatingState.rideId) (state.data.rideRatingState.feedbackList))
-      _ <- Remote.rideFeedbackBT (Remote.makeFeedBackReq (state.data.rideRatingState.rating) (state.data.rideRatingState.rideId) (state.data.rideRatingState.feedback))
+      _ <- Remote.rideFeedbackBT (Remote.makeFeedBackReq (state.data.rideRatingState.rating) (state.data.rideRatingState.rideId) (state.data.rideRatingState.feedback) (state.data.ratingViewState.wasOfferedAssistance))
```
[the remark this answers is not in the document]
Comment (Shikhar-24) [reply Frontend/ui-customer/src/Flow.purs]: done

**Comment by kavya-shree-s** [comment]
@Shikhar-24 Can you attach screenshot or screen recording here

**Comment by Shikhar-24** [comment]
video link added  @kavya-shree-s

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**